docusaurus/docs/tutorials/convert-a-frontend-datasource-to-backend.md

```diff
@@ -0,0 +1,411 @@
+---
+id: convert-a-frontend-datasource-to-backend
+title: How to convert a frontend data source plugin into a backend plugin
+sidebar_position: 10
+description: Learn how to convert a frontend data source plugin into a backend plugin
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - backend
+  - frontend
+  - datasource
+---
+
+# How to add a frontend data source to a backend component
+
+This guide shows you how to convert an existing frontend-only data source plugin into a [backend plugin](https://grafana.com/developers/plugin-tools/introduction/backend-plugins).
+
+To convert the frontend data source, we recommend scaffolding a new backend data source plugin using [`npx @grafana/create-plugin@latest`](https://grafana.com/developers/plugin-tools/). Use the following instructions to extend this foundation to copy functionality from your original plugin.
+
+## Before you begin
+
+Before you dive into the details, you should familiarize yourself with the process of creating a backend data source plugin. If you haven't done this before, you can follow our tutorial for [building a backend plugin](https://grafana.com/developers/plugin-tools/tutorials/build-a-data-source-backend-plugin).
+
+## Key concepts
+
+Before going into specific conversion advice, we will discuss the main components of a data source and how these differ between frontend and backend plugins.
+
+### Frontend `DataSource` class
+
+Data source plugins implement a new `DataSourcePlugin`. This class takes as a parameter a `DataSource` class, which for frontend data sources extends `DataSourceApi`, and for backend data sources extends `DataSourceWithBackend`. Because the `DatasourceWithBackend` class already implements most of the required methods, you can migrate to it to significantly simplify your code.
+
+Data source plugins require two components: a query editor and a config editor.
+
+**Examples:**
+
+- [Frontend data source](https://github.com/grafana/grafana-plugin-examples/blob/main/examples/datasource-http/src/DataSource.ts#L14).
+- [Backend data source](https://github.com/grafana/grafana-plugin-examples/blob/main/examples/datasource-http-backend/src/datasource.ts#L6).
+
+### Query and config editor
```

Comment:
FTR, I have merged the query and config editor sections into one since it was quite similar.